lib/message/context.ts

```diff
@@ -0,0 +1,102 @@
+/**
+ * The context is an abstraction we use to add domain separation to OpenPGP signatures.
+ * It's implemented by adding notation data to signatures, which may be marked as critical, so that
+ * the resulting signature can only be verified by a verifier who expects the context to be present.
+ */
+import { SignaturePacket, RawNotation, PartialConfig } from '../openpgp';
+import { normalizeDate } from './utils';
+
+// Different contexts will affect the notation's value, not the name
+const CONTEXT_NOTATION_NAME = 'context@proton.ch';
+
+export interface ContextSigningOptions {
+    /** context identifier */
+    id: string,
```

Comment:
I would call this `value` rather than `id`, even if it's a bit generic, it really could be anything in principle, not just an ID; e.g. in the case of Mail we could put some headers there, etc.